lib/Doctrine/ORM/Cache/DefaultQueryCache.php

```diff
@@ -260,7 +260,7 @@ public function put(QueryCacheKey $key, ResultSetMapping $rsm, $result, array $h
             throw new CacheException("Second-level cache query supports only select statements.");
         }
 
-        if (isset($hints[Query::HINT_FORCE_PARTIAL_LOAD]) && $hints[Query::HINT_FORCE_PARTIAL_LOAD]) {
```

Comment:
I have never used this hint but is there any context where it would be used and, with this change, would start caching things in the L2C?

Reply:
You are right, that might be the case. Both hints need to lead to exception.